pytorch_lightning/utilities/apply_func.py

```diff
@@ -245,7 +245,11 @@ def batch_to(data):
             return device_data
 
         kwargs = dict(non_blocking=True) if isinstance(data, torch.Tensor) else {}
-        return data.to(device, **kwargs)
+        data_output = data.to(device, **kwargs)
+        if data_output is not None:
+            return data_output
+        # user wrongly implemented the ``TransferableDataType`` and forgot to return ``self``.
```

Comment:
nit: double backticks are for rendered docs

```suggestion
        # user wrongly implemented the `TransferableDataType` and forgot to return `self`.
```